src/Protocol/ActivityPub.php

```diff
@@ -0,0 +1,1919 @@
+	 */
+	private static function switchContacts($receivers, $actor)
+	{
+		if (empty($actor)) {
+			return;
+		}
+
+		foreach ($receivers as $receiver) {
+			$contact = DBA::selectFirst('contact', ['id'], ['uid' => $receiver, 'network' => Protocol::OSTATUS, 'nurl' => normalise_link($actor)]);
+			if (DBA::isResult($contact)) {
+				self::switchContact($contact['id'], $receiver, $actor);
+			}
+
+			$contact = DBA::selectFirst('contact', ['id'], ['uid' => $receiver, 'network' => Protocol::OSTATUS, 'alias' => [normalise_link($actor), $actor]]);
+			if (DBA::isResult($contact)) {
+				self::switchContact($contact['id'], $receiver, $actor);
+			}
+		}
+	}
+
+	/**
+	 * @brief 
+	 *
+	 * @param $object_data
+	 * @param array $activity
+	 *
+	 * @return 
+	 */
+	private static function addActivityFields($object_data, $activity)
+	{
+		if (!empty($activity['published']) && empty($object_data['published'])) {
+			$object_data['published'] = $activity['published'];
+		}
+
+		if (!empty($activity['updated']) && empty($object_data['updated'])) {
+			$object_data['updated'] = $activity['updated'];
+		}
+
+		if (!empty($activity['inReplyTo']) && empty($object_data['parent-uri'])) {
+			$object_data['parent-uri'] = JsonLD::fetchElement($activity, 'inReplyTo', 'id');
+		}
+
+		if (!empty($activity['instrument'])) {
+			$object_data['service'] = JsonLD::fetchElement($activity, 'instrument', 'name', 'type', 'Service');
+		}
+		return $object_data;
+	}
+
+	/**
+	 * @brief 
+	 *
+	 * @param $object_id
+	 * @param $object
+	 * @param $trust_source
+	 *
+	 * @return 
+	 */
+	private static function fetchObject($object_id, $object = [], $trust_source = false)
+	{
+		if (!$trust_source || is_string($object)) {
+			$data = self::fetchContent($object_id);
+			if (empty($data)) {
+				logger('Empty content for ' . $object_id . ', check if content is available locally.', LOGGER_DEBUG);
+				$data = $object_id;
+			} else {
+				logger('Fetched content for ' . $object_id, LOGGER_DEBUG);
+			}
+		} else {
+			logger('Using original object for url ' . $object_id, LOGGER_DEBUG);
+			$data = $object;
+		}
+
+		if (is_string($data)) {
+			$item = Item::selectFirst([], ['uri' => $data]);
+			if (!DBA::isResult($item)) {
+				logger('Object with url ' . $data . ' was not found locally.', LOGGER_DEBUG);
+				return false;
+			}
+			logger('Using already stored item for url ' . $object_id, LOGGER_DEBUG);
+			$data = self::createNote($item);
+		}
+
+		if (empty($data['type'])) {
+			logger('Empty type', LOGGER_DEBUG);
+			return false;
+		}
+
+		if (in_array($data['type'], self::CONTENT_TYPES)) {
+			return self::processObject($data);
+		}
+
+		if ($data['type'] == 'Announce') {
+			if (empty($data['object'])) {
+				return false;
+			}
+			return self::fetchObject($data['object']);
+		}
+
+		logger('Unhandled object type: ' . $data['type'], LOGGER_DEBUG);
+	}
+
+	/**
+	 * @brief 
+	 *
+	 * @param $object
+	 *
+	 * @return 
+	 */
+	private static function processObject(&$object)
```

Comment:
Typo? Is the `&` before `$object` an error?

Reply:
Although nothing in this method changes its value, so it's superfluous.